lib/proxy/router.go

```diff
@@ -280,6 +363,53 @@ func (r *Router) DialHost(ctx context.Context, from net.Addr, host, port, cluste
 	return newProxiedMetricConn(conn), trace.Wrap(err)
 }
 
+func preformRBACCheck(ctx context.Context, idInfo IdentityInfo, cluster, targetCluster string, client auth.ClientI, target types.Server) error {
```

Comment:
This is basically a copy of `canLoginWithRBAC`. Can we please keep the login in one place? If some changes to the logic are needed you can probably pass a function/interface as a function parameter instead of copying the whole function.